.babelrc

```
@@ -7,6 +7,14 @@
           "node": "6"
         }
       }
+    ],
+    [
+      "@babel/preset-typescript",
+      {
+        "allowNamespaces": true,
```

```suggestion
        "allowNamespaces": false,
```

legacy stuff, let's skip it

---

Ok I'll disable it. I thought I should enable it because it would be [enabled by default in babel 8](https://babeljs.io/docs/en/babel-plugin-transform-typescript#allownamespaces)

---

sure, but https://github.com/microsoft/TypeScript/issues/30994#issuecomment-492017219

---

The flag is now disabled. But as a side note, I think there are still use cases for ts namespaces -- to provide type definitions for js libraries that requires using namespace merging to augment values, and to group type/interface definitions so they can be accessed with a dot (`SomeNamespace.SomeType`).

But I think for ts code that would emit js, it's best to not use the namespace feature (since it is not tree-shakable if used with a bundler, emits more boilerplate code and is not supported by compilers other than `tsc`), so turning off the `allowNamespaces` flag would be a more sensible choice so no code written with `namespace` code would be accidentally emitted.